File: packages/sdks/server-sdks/gatrix-dotnet-server-sdk/src/Gatrix.Server.Sdk/Evaluation/Strategies/DefaultStrategy.cs
namespace Gatrix.Server.Sdk.Evaluation.Strategies;

/// <summary>
/// Default strategy — always returns true.
/// Only constraints and segments determine the outcome.
/// </summary>
public class DefaultStrategy : IStrategy
{
    public string Name => "default";

    public bool IsEnabled(Models.StrategyParameters? parameters, Models.EvaluationContext context, string flagName)
    {
        return true;
    }
}

File: packages/evaluator/src/strategies/DefaultStrategy.ts
/**
 * Default Strategy - always returns true
 * Only constraints and segments determine the outcome.
 */

import { EvaluationContext, StrategyParameters, StrategyEvaluationResult } from '@gatrix/shared';
import { Strategy } from './Strategy';

export class DefaultStrategy extends Strategy {
  constructor() {
    super('default');
  }

  isEnabled(_parameters: StrategyParameters, _context: EvaluationContext): boolean {
    return true;
  }

  isEnabledWithDetails(
    _parameters: StrategyParameters,
    _context: EvaluationContext
  ): StrategyEvaluationResult {
    return { enabled: true, reason: 'Default strategy always enabled' };
  }
}
